packages/fxa-settings/src/pages/Signup/container.spec.tsx

```diff
@@ -0,0 +1,439 @@
+    .mockImplementation(() => {});
+}
+
+// TIP - Sometimes it's useful to have inner mocks. In this case, we hold a reference to
+// mockBeginSignupMutation, so it can be asserted against and redefined as needed.
+let mockBeginSignupMutation = jest.fn();
+function mockApolloClientModule() {
+  mockBeginSignupMutation.mockImplementation(async () => {
+    return {
+      data: {
+        unwrapBKey: 'foo',
+        SignUp: {
+          uid: 'uid123',
+          keyFetchToken: 'kft123',
+          sessionToken: 'st123',
+        },
+      },
+    };
+  });
+
+  jest.spyOn(ApolloModule, 'useMutation').mockReturnValue([
+    async (...args: any[]) => {
+      return mockBeginSignupMutation(...args);
+    },
+    {
+      loading: false,
+      called: true,
+      client: {} as ApolloClient<any>,
+    },
+  ]);
+}
+
+// TIP - Occasionally, due to how a module is constructed, jest.spyOn will not work.
+// In this case, use the following pattern. The jest.mock approach generally works,
+// but as you can see, it's quite a bit noisier.
+jest.mock('../../models', () => {
+  return {
+    ...jest.requireActual('../../models'),
+    useAuthClient: jest.fn(),
+  };
+});
+function mockModelsModule() {
+  let mockAuthClient = new AuthClient('localhost:9000');
+  mockAuthClient.accountStatusByEmail = jest
+    .fn()
+    .mockResolvedValue({ exists: true });
+  (ModelsModule.useAuthClient as jest.Mock).mockImplementation(
+    () => mockAuthClient
+  );
+}
+
+// TIP - Finally, we should create a helper function, so the defacto
+// mock behaviors can be easily applied. Once applied, they can
+// always be overridden as needed.
+function applyMocks() {
+  // TIP - Important! Clear previous mock states
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+
+  // Run default mocks
+  mockIntegration();
+  mockServiceName();
+  mockSignupModule();
+  mockApolloClientModule();
+  mockModelsModule();
+  mockUseValidateModule();
+  mockFirefoxModule();
+  mockCryptoModule();
+  mockReachRouterModule();
+  mockReactUtilsModule();
+  mockLoadingSpinnerModule();
+}
+
+// TIP - Since render looks more or less the same each time, we can tease this out
+// as a helper function.
+async function render(text?: string) {
+  // Even though we aren't testing presentation, there still might be presentation
+  // state/context driven by react or l10n that's required. Therefore we will invoke the
+  // container component as follows.
+  renderWithLocalizationProvider(
+    <SingupContainer
+      {...{
+        integration,
+        serviceName,
+      }}
+    />
+  );
+
+  // TIP - Wait for the expected mocked test to show up.
+  await screen.findByText(text || 'sign up mock');
+
+  // TIP/HACK -To work around the fact that the container uses, requestAnimationFrame...
+  // Turns out that can create some pretty erratic test behavior...
+  await new Promise((r) => setTimeout(r, 100));
+}
+
+// TIP - It's easier to target test suites that don't have spaces in their names.
```

Comment:
Is this a convention we're wanting to set across the codebase? I would expect to call this one at least "SignupContainer" to match the exported component name, but maybe kebab case has an advantage? I think I'm mostly not sure how useful this is for "inner" describes, but I'm open to it and would want us to start being consistent moving forward.

Reply:
I might be nice, at least for the describes... Even jest's documented example use of `-t` have dashes in spec names. If one runs, `jest -t 'some test'`, jest converts this to `some|test`, which ends up testing to much. You can provide an exact regex, so you could `jest -t test\stest` but that's kinda annyoing...